### ISSUE TITLE
Move to github.com/ko-build

### ISSUE THREAD

**Issue by imjasonh** [google/ko#791]
An incomplete list of steps, more or less in order:

- [x] create github.com/ko-build org
  - [ ] move this repo to github.com/ko-build (early September)
- [ ] https://github.com/google/ko/pull/749
  - [ ] set up `ko.build` vanity importpath redirect using `<meta>` tag
- [ ] Go module name change
  - [ ] cut v0.12, the last release as github.com/google/ko
  - [ ] [deprecate](https://go.dev/ref/mod#go-mod-file-module) `google/ko@v0.12.0` and point users to `ko.build@v0.13`
  - [ ] rename Go module in `go.mod` to `module ko.build`
  - [ ] release v0.13 of `ko.build`
- [ ] `ghcr.io/google/ko` image name change
  - ⚠️  GitHub will apparently _not_ automatically redirect package requests; users will need to update their references to get future updates
- GitHub Releases download links _will_ apparently redirect, _so long as https://github.com/google/ko is not recreated later!_
- [ ] rename K8s slack channel `#ko-project` to `#ko-build` ([ref](https://github.com/kubernetes/community/compare/master...imjasonh:community-1:ko-build))
- [ ] move https://github.com/imjasonh/setup-ko
  - [ ] update `setup-ko` to fetch released binaries from https://github.com/ko-build/ko/releases
  - [ ] cut v0.6
  - GitHub will apparently redirect requests for `imjasonh/setup-ko`
- [ ] https://github.com/chainguard-dev/terraform-provider-ko
  - [ ] update TF module release to point to new location, in new TF org

cc @mchmarny 

If anybody thinks of anything that's missing, please let us know!

**Comment by mchmarny** [comment]
Thanks for putting this together, Jason. 

The Go releases section is little unclear. Are you saying that the v0.12 release will have to be done BEFORE the repo is moved or after... because this is all just go mod magic anyway and we can do this transparently under `github.com/ko-build/ko`? 

Also, the lack of redirects on image pull requests is disappointing but kind of makes sense. Do we have a feel for the volume of unique pull sources we see on daily bases? It would be good to capture some metrics around this to understand the success rate of the image migration.

**Comment by imjasonh** [comment]
> The Go releases section is little unclear. Are you saying that the v0.12 release will have to be done BEFORE the repo is moved or after... because this is all just go mod magic anyway and we can do this transparently under `github.com/ko-build/ko`?

Good question. When the repo is moved, it will still have the Go module name `github.com/google/ko` because that's what's in the [`go.mod`](https://github.com/google/ko/blob/6bda5508996591110959c0d68c6013d120e0ae23/go.mod), and because GitHub will redirect the repo URL for us.

We can cut v0.12 any time, up to today even, with that notice that it will be moved in the future. I'd like to do this release first because I don't want the `google/ko` -> `ko.build` move to include any functional changes, and there's a lot of functional changes since v0.11 in March.

> Also, the lack of redirects on image pull requests is disappointing but kind of makes sense. Do we have a feel for the volume of unique pull sources we see on daily bases? It would be good to capture some metrics around this to understand the success rate of the image migration.

The image doesn't get that much usage, honestly. 

https://github.com/google/ko/pkgs/container/ko indicates it's been downloaded 3.7k times since it was added in December (https://github.com/google/ko/pull/528), which is 15 pulls/day on average.

You can see per-version pull rates at https://github.com/google/ko/pkgs/container/ko/versions -- clicking back through the last few pages I don't see any that got more than ~150 pulls.

**Comment by mchmarny** [comment]
Thanks, +1 on the version we cut in the new repo not having any functional changes. Keeps things clean. 

On image pull metrics, I guess the daily doesn't mean much as some use cases may not be pulling as frequently or use caching which further reduces the cadence. Still, there isn't much we can do here so giving ample of notice (which we do) and assisting people with pointers to the new image when ready, is the best we can do.

**Comment by imjasonh** [comment]
For folks watching this, this broke certain usages of `setup-ko`, which are fixed in https://github.com/imjasonh/setup-ko/pull/4

This only broke users who relied on the implicit default `latest-release`, and users who specified a `version` (including `tip`) should be unaffected.

You can update to use this commit:

```
- uses: imjasonh/setup-ko@ace48d793556083a76f1e3e6068850c1f4a369aa
```

**Comment by mchmarny** [comment]
I've enable public packages on the repo level which enabled us to make the ko package public. Let me know @imjasonh if you see any more issues with this

**Comment by imjasonh** [comment]
Just an update: most of the items remaining are waiting on the outcome of ko's proposal as a CNCF sandbox project, and making myself and any other potential non-Google contributors org owners so we can carry out the remaining items. When (if) CNCF accepts the project we'll also donate the ko.build domain, which I currently personally own, and as such don't feel comfortable making it a load-bearing part of ko's infrastructure until its bus factor is a little higher.

tl;dr: remaining items currently blocked on CNCF Sandbox decision. 🤞

**Comment by BenTheElder** [comment]
https://opensource.googleblog.com/2022/10/ko-applies-to-become-a-cncf-sandbox-project.html

=>

https://github.com/cncf/toc/pull/945

=> 

https://github.com/cncf/toc/pull/945#issuecomment-1282644808

=>

https://sandbox.cncf.io/

I don't see a currently open Sandbox application anywhere, I think there may be a gap there?

**Comment by imjasonh** [comment]
Thanks @BenTheElder that does indeed seem like a gap. 🙃 

@mchmarny Did you open a Sandbox application through the sandbox repo after toc#945 was closed?

I'll start one now and link it here.

**Comment by imjasonh** [comment]
Just an update: https://github.com/imjasonh/setup-ko has moved to the `ko-build` org.

The last two remaining items at this point are to move the terraform provider, and to update the importpath. I hope to move the terraform provider today, and the importpath possibly next week.
